### PR TITLE
Dropped calls to obsolete NetworkInterfaces API

### PR DESCRIPTION
**Pull request by mchf**
bnc#964856

### CODE REVIEW

File: src/include/network/lan/address.rb [hunk 1]
```diff
@@ -1430,9 +1424,7 @@ def AddressDialog
       end
 
       # proceed with WLAN settings if appropriate, #42420
-      if ret == :next && LanItems.type == "wlan" && LanItems.alias == ""
```
Comment (teclator) [inline src/include/network/lan/address.rb]: So, aliases are not important anymore, could it arise in some problem?

**Comment by teclator** [comment]
Drop code seems to be OK, but should be fine to fix real bug during the cleanup process

File: src/include/network/lan/address.rb (hunk 1 above)
Comment (mchf) [reply src/include/network/lan/address.rb]: ```LanItems::aliases``` is still in game. ```LanItems::alias``` is not. Looks confusing? That happens when someone re-implements a new feature and do not clean the code properly.

**Comment by mchf** [comment]
It is a real bug ... see description ;-) but seriously using obsolete api *is* a bug and one never knows when it bits him. And we do not wait for a customer's report. That it hasn't crashed in this case up to know is partly luck and partly a result of nontrivial effort spent in various other bugs.

**Comment by teclator** [comment]
Code looks good, but I missed like unit tests or some explanations, although taking in account that it is probably leftlover code then it is ok

File: src/include/network/lan/cmdline.rb [hunk 2]
```diff
@@ -194,9 +194,9 @@ def AddHandler(options)
         [LanItems.current, "ifcfg"],
         Ops.get(options, "name", "")
       )
-      LanItems.type = NetworkInterfaces.device_type(
-        Ops.get(options, "name", "")
-      )
+      LanItems.type = options["type"]
+      raise "Device type is mandatory." if !LanItems.type
```
Comment (jreidinger) [inline src/include/network/lan/cmdline.rb]: this looks like API change to me. So now in CLI type is mandatory? If so, then please mention it in changes file

Comment (mchf) [reply src/include/network/lan/cmdline.rb]: yes. Bcs before it was "detected" from device name which is not possible anymore. In UI it is done by explicitly selecting by user ... so I had to introduce something similar here too.

Will add change log entry.

File: package/yast2-network.changes [hunk 3]
```diff
@@ -3,6 +3,8 @@ Thu Jul 12 11:46:53 UTC 2018 - mfilka@suse.com
 
 - bnc#964856
   - dropped calls to obsolete NetworkInterfaces API
+  - mentioning device type explicitly using new option is mandatory
+    since now when adding device using CLI
```
Comment (jreidinger) [inline package/yast2-network.changes]: I would rephrase it a bit: "CLI: for new devices the option type is now mandatory"
